test/expect-tests/dune_rpc/dune_rpc_tests.ml

```diff
@@ -925,3 +925,37 @@ let%expect_test "print digests for all public RPCs" =
       Request: Sexp
       Response: 33528f248084297d123a6ebd4c3ddee0 |}]
 ;;
+
+let () =
+  Stdune.Path.set_root (Stdune.Path.External.of_filename_relative_to_initial_cwd ".")
+;;
+
+let test ?dir ?promotion main =
+  let id = Dune_engine.Build_system.Error.Id.gen () in
+  let description =
+    `Diagnostic (Dune_engine.Compound_user_error.make ~main ~related:[])
+  in
+  Dune_console.printf "---- Original ----";
+  Dune_console.print_user_message main;
+  Dune_console.printf "------- RPC ------";
+  Dune_engine.Build_system.Error.For_tests.make ~id ~description ~dir ~promotion ()
+  |> Dune_rpc_impl.Diagnostics.For_tests.diagnostic_of_error
+  |> Dune_rpc_private.Diagnostic.to_user_message
+  |> Dune_console.print_user_message
+;;
+
+let%expect_test "serialize and deserialize error message with location" =
+  let loc = Stdune.Loc.of_pos ("Bar", 1, 2, 3) in
+  let dir = Stdune.Path.of_string "/Foo" in
+  test ~dir (User_error.make ~loc [ Pp.verbatim "An error with location!" ]);
+  [%expect
+    {|
+    ---- Original ----
+    File "Bar", line 1, characters 2-3:
+    Error: An error with location!
+    ------- RPC ------
+    Error: (In directory /Foo)
+    File "/Foo/Bar", line 1, characters 2-3:
+
+    Error: An error with location! |}]
```

Comment:
@rgrinberg It seems to me like we shouldn't have to print the `In directory` part as the loc already contains this info. WDYT?

Reply:
Yeah, I guess it's redundant if the location is already there.